src/server/twemcache-rs/src/protocol/data/response.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+//! Responses which are sent back to a client
+
+use bytes::BytesMut;
+use segcache::Item;
```

Comment:
abstraction leak. Can we just keep it at `(key, value)` tuple(s)? This prevents directly depending on storage

Reply:
It's a little more than that, but we can unpack the relevant fields into the struct-like enum variant and do things like preconvert the flags from the variable length optional data into a u32. Fix will be in the next push to the branch